schemas/tests.yaml

```diff
@@ -0,0 +1,158 @@
+---
+
+#
+# JSON Schema definition for tmt `Tests`
+# https://tmt.readthedocs.io/en/stable/spec/tests.html
+#
+
+definitions:
+
+  beakerlib_library:
+    type: array
+    minItems: 1
+
+    items:
+      oneOf:
+        # library(repo/lib)
+        - type: string
+          pattern: "^library([a-zA-Z0-9]+/[a-zA-Z0-9]+)"
+
+        # url: https://github.com/beakerlib/openssl
```

Comment:
* I believe the following definitions - with the exception of the one with `nick` field - are not just "Beaker library reference", but also more generic "fmf identifier" structure that's used frequently in other places. I'd propose defining this fmf id, then branching the `nick` special case of from it, because, sooner or later, we will need fmf id definition to exist anyway (`link`, for example).
* if I'm reading fmf docs right, all keys are optional. Which is probably unfortunate when it comes to schema, but I think jsonschema supports requiring at least one of the object properties to be present.

So something like this could be a good start for the fmf id:

```yaml
# https://fmf.readthedocs.io/en/stable/concept.html#identifiers
- type: object

  properties:
    url:
      type: string
      # maybe also "format: uri" together with "pattern" to enforce http instead of all possible protocols?
      pattern: "^https?://"
    ref:
      type: string
    path:
      type: string
    name:
      type: string
      pattern: "^/.+"

  minProperties: 1
```

There is some limitation WRT `additionalProperties: false`, https://json-schema.org/understanding-json-schema/reference/object.html?highlight=minproperties#extending-closed-schemas, I'd maybe call this as sort of "fmf id core", then wrap it with "fmf id" which would add `additionalProperties: false` and add a second "wrapper" called "beakerlib" which would add its own `nick` field and also `additionalproperties: false`.

Reply:
will leave this one for tomorrow to scratch my head around it :)

Reply:
I will need to add some examples so I can start validating, I am not even sure the schema now validates anything :)

Reply:
There are plenty of tests in the TMT repo O;)

Reply:
agreed, testing ...

Reply:
Or (internal) tests/python, tests/ruby...)

Reply:
Or upstream like LLVM suite (https://src.fedoraproject.org/rpms/llvm/blob/rawhide/f/tests)